aten/src/ATen/native/cpu/PowKernel.cpp

```diff
@@ -34,126 +36,112 @@ void pow_tensor_tensor_kernel(TensorIteratorBase& iter) {
-            return 1.0 / std::sqrt(base);
-          },
-          [](Vec base) -> Vec { return base.rsqrt(); }
-        );
-      } else if (exp == -1) {
-        cpu_kernel_vec(iter,
-          [](scalar_t base) -> scalar_t {
-            return 1.0 / base;
-          },
-          [](Vec base) -> Vec { return base.reciprocal(); }
-        );
-      } else if (exp == -2) {
-        cpu_kernel_vec(iter,
-          [](scalar_t base) -> scalar_t {
-            return 1.0 / (base * base);
-          },
-          [](Vec base) -> Vec { return (base * base).reciprocal(); }
-        );
-      } else {
-        cpu_kernel_vec(iter,
+      cpu_kernel_vec(iter,
           [=](scalar_t base) -> scalar_t {
             return std::pow(base, exp);
           },
           [=](Vec base) -> Vec { return base.pow(exp); }
-        );
-      }
-    });
-  } else if (isComplexType(iter.dtype())) {
-    const auto exp = exp_scalar.to<c10::complex<double>>();
-    // Floating types allow AVX2 vector optimizations for pow/sqrt/rsqrt:
-    AT_DISPATCH_COMPLEX_TYPES(iter.dtype(), "pow", [&]() {
-      using Vec = Vec256<scalar_t>;
-      if (exp == 0.5) {
-        cpu_kernel_vec(iter,
-          [](scalar_t base) -> scalar_t {
-            return std::sqrt(base);
-          },
-          [](Vec base) -> Vec { return base.sqrt(); }
-        );
-      } else if (exp == 2.0) {
-        cpu_kernel_vec(iter,
-          [](scalar_t base) -> scalar_t {
-            return base * base;
-          },
-          [](Vec base) -> Vec { return base * base; }
-        );
-      } else if (exp == 3.0) {
-        cpu_kernel_vec(iter,
-          [](scalar_t base) -> scalar_t {
-            return base * base * base;
-          },
-          [](Vec base) -> Vec { return base * base * base; }
-        );
-      } else if (exp == -0.5) {
-        cpu_kernel_vec(iter,
-          [](scalar_t base) -> scalar_t {
-            return scalar_t(1.0) / std::sqrt(base);
-          },
-          [](Vec base) -> Vec { return base.rsqrt(); }
-        );
-      } else if (exp == -1.0) {
-        cpu_kernel_vec(iter,
-          [](scalar_t base) -> scalar_t {
-            return scalar_t(1.0) / base;
-          },
-          [](Vec base) -> Vec { return base.reciprocal(); }
-        );
-      } else if (exp == -2.0) {
-        cpu_kernel_vec(iter,
-          [](scalar_t base) -> scalar_t {
-            return scalar_t(1.0) / (base * base);
-          },
-          [](Vec base) -> Vec { return (base * base).reciprocal(); }
-        );
-      } else {
-        cpu_kernel_vec(iter,
-          [=](scalar_t base) -> scalar_t {
-            return std::pow(base, scalar_t(exp));
-          },
-          [=](Vec base) -> Vec { return base.pow(scalar_t(exp)); } // std::pow cannot accept mixed complex data types.
-        );
-      }
-    });
+      );
+    }();
+  } else if (dtype == ScalarType::BFloat16) {
+      using scalar_t =
```

Comment:
@imaginary-person This is no longer needed and is causing errors in our internal builds. Can you remove it please.

Reply:
Oops, I had missed it while copy,pasting the AT_DISPATCH_FLOATING_TYPES_AND macro. Thanks for catching it!

Reply:
@heitorschueroff, the CI failures are unrelated. I can rebase if you'd like. Thanks!